SalesforceCommerceCloud/pwa-kit
# pwa-kit-dev command for tailing logs

Adds the ability to tail logs for deployed Managed Runtime environments!

# Description



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes
- Incorporate Oli's `managedRuntimeCommand` [helper](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/763/files#diff-52cf52d1c1e84c76145d9b0a69946921f732dbff0fa9a1295946093b0fe2779cR93) into the pwa-kit-dev CLI to make sharing common options like `cloudOrigin` and `credentialsFile` easier, and use it for `push` and `save-credentials`
- Add a `tail-logs` command
- Add a `tail-logs` npm script to each template's package.json
- Add/update tests
- Upgrade `commander`'s major version

# How to Test-Drive This PR
- Ensure your production API key is in `~/.mobify`
- Checkout this branch
- If you haven't done so already, install [lerna](https://www.npmjs.com/package/lerna), then run `npm ci` at the root of the repo
- Tail logs! Run `node packages/pwa-kit-dev/bin/pwa-kit-dev.js tail-logs -p <slug> -e <slug>` for an environment in production. Once you generate some traffic on the environment, you should see something like this:

  <img width="777" alt="Screen Shot 2022-11-01 at 11 17 43 AM" src="https://user-images.githubusercontent.com/21148709/199308310-e8d284bb-cb7e-45ad-acc5-2f2f2f501667.png">
- Other things to try:
  - Make sure the command's help text includes the project, environment, cloud origin, and credentials file options
  - Start tailing, but don't open the environment URL until > 10 minutes later (the idle timeout). The connection should remain open and stream the logs once you open it.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
